Added DotCover Cover command

mholo65 opened the pull request:
Added DotCover Cover command. Needed since TeamCity only want snapshots, not HTML/XML reports.

If possible, to remove duplicate code, I would like to reuse **DotCoverCoverSettings** in **DotCoverAnalyser** by having **DotCoverAnalyseSettings** to inherit from **DotCoverCoverSettings**.  Is this allowed since it will make documentation unclear?

NOTE: Quickly hacked using my Linux box, need to verify functionality on Windows (dotCover.exe doesn't run under Mono). But this have to wait until monday when I'm back at work.

//cc @patriksvensson

patriksvensson commented:
@mholo65 Nice! :+1:

Funny, but as you published this PR I was updating the site to show inherited members better (https://github.com/cake-build/website/pull/65) :smile: As soon as that PR is merged, this won't be a problem anymore. As long as it's not a breaking change, the refactoring is OK with me.

I would however recommending adding an abstract `DotCoverSettings` for settings shared between different (related) DotCover tools and add all shared functionality there.

Great job!

mholo65 commented:
Nice! Shouldn't be a breaking change, **Analyse** uses all settings as **Coverer** plus **ReportType**.

mholo65 commented:
@patriksvensson, refactored, rebased and done.

An additional suggestion for removing duplicate code. 

- Create an abstract class for handling **DotCoverSettings** argument creation and interception of the unit test runner process.
```csharp
DotCoverWhateverName<TSettings> : DotCoverTool<TSettings> where TSettings : DotCoverSettings
```
- Or an **internal** extension method (GetArguments) for **DotCoverSettings** and move InterceptAction to **DotCoverTool**
```csharp
internal static ProcessArgumentBuilder GetArguments(this DotCoverSettings settings, ProcessArgumentBuilder builder)
```
What do you think? Or should we leave it as is.

patriksvensson commented:
@mholo65 Sorry for getting back sooner about this.
This PR looks good to me :+1: 

I think it's better to do more refactoring in a separate PR. It makes it easier to see how things are affected.

@devlead and @gep13: Any comments on this PR before merging?

devlead commented on src/Cake.Common/Tools/DotCover/Analyse/DotCoverAnalyseSettings.cs:
```diff
@@ -23,56 +19,10 @@ public sealed class DotCoverAnalyseSettings : ToolSettings
-
-        /// <summary>
-        /// Gets the coverage filters using the following syntax: +:module=*;class=*;function=*;
-        /// Use -:myassembly to exclude an assembly from code coverage. 
```
Trim trailing white space here.

devlead commented:
@patriksvensson looks good to me, just left one _*minor*_ comment, remove WIP and rebase, also do we have an referring issue to the PR assigned to a release?

patriksvensson commented:
@devlead Doh! Assumed there was an issue for this.

@mholo65 Can you create an issue for this?

patriksvensson commented:
@mholo65 Perfect, thanks!

mholo65 commented:
Trimmed, rebased and done.

@devlead You probably meant the trailing whitespace in *DotCoverSettings.cs:35* not *DotCoverAnalyseSettings.cs:35* since You are referring to an deletion in your comment.

mholo65 commented:
HOLD!. sorry, forgot to rebase :)

mholo65 commented:
Rebased and checks have passed.

patriksvensson commented:
Merged! Thank you for this! :+1:
